tabular/src/autogluon/tabular/models/xgboost/xgboost_model.py

```diff
@@ -54,13 +55,14 @@ def get_eval_metric(self):
     def _preprocess(self, X, is_train=False, max_category_levels=None, **kwargs):
         X = super()._preprocess(X=X, **kwargs)
 
-        if self._ohe_generator is None:
-            self._ohe_generator = xgboost_utils.OheFeatureGenerator(max_levels=max_category_levels)
-
         if is_train:
-            self._ohe_generator.fit(X)
+            if self._ohe:
+                if self._ohe_generator is None:
```

Comment:
If the model is re-trained (i.e. on the full data), wouldn't this reuse the generator. I think it's better to always create a new one if `is_train and self._ohe`.

Reply:
No, re-training would be a new class instance, it does not cause issues